### PR TITLE
Splits Space Adapt into two

### PR DESCRIPTION
**Pull request by Poivre-Vert**
## About The Pull Request

Splits cold resistance off of space adapt and into firey sweat, and also fixes an unintended issue with the DNA console being able to print an unlimited number of chromosomes in a matter of seconds.

## Why It's Good For The Game

For a while now the geneticist meta has essentially to rush space adapt and distribute as many as possible to the crew, causing a lot of areas, like EVA, to be redundant, not to mention that the average crew member does not need to be space proof less than 10 minutes into the round.

By making true space proofing through genetics have a real downside to it (as well as slowing down the process to get it) will force geneticists to work with science or virology to off set the downside of firey sweat, or to ignore it all together by working with other departments to begin with.

Fixing the infinite chromosome exploit will also make geneticists have to interact with crew using their DNA scanner and DNA activators rather than sitting in the pod using tk to spam one button.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>
I'll have to add a screen shot in after the fact as I'm not used to VSC, but for the most part I'd need videos to show off the changes and that they work and all that.

[https://media.discordapp.net/attachments/1014958653578678275/1062952466980814918/image.png](url)

[https://media.discordapp.net/attachments/1014958653578678275/1062952663999844392/image.png](url)
Bare computers after trying to do the glitch, also just having everything on me etc.

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
balance: Split space adapt between itself and firey sweat
fix: fixed infinite chromosomes via DNA console
/:cl:

### CODE REVIEW

**Comment by Fronsis** [comment]
Screenshot doesn't seem to work

I don't think it's a good idea to give the cold resistance to a mutation that constantly makes you catch on fire, it's already a pain in the ass because your stuff get burnt and this is why it's a noob trap for a lot of new geneticists because they think that mutations woul make them at least resistant to fire

I think /TG/ has a weird version of their own space adapt but i think it's more... pressure related? i've seen it in a couple of videos.

**Comment by Poivre-Vert** [comment]
> they think that mutations woul make them at least resistant to fire

It actually does make you pretty resistant to fire, it's oozeling level fire resistance.

Anyways the point of it is to make genetics work with other departments, and the cold damage is very minimal without firey sweat, so you can either use nanites to maintain your body temperatureand heal the damage you take, or get a virus made that will make you not catch fire if you decide to use sweat. Could even work with the chaplain or xenobiolgist if you just want to be constantly immolated.

**Comment by EvilDragonfiend** [comment]
That's what TG made. I think you might want to check their code.

**Comment by Poivre-Vert** [comment]
TG has pressure adapt and temperature adapt, which combined end up being stronger than our current space adapt, I don't think giving genetics a free atmos hard suit is a great idea, because as it stands the only thing space adapt is currently weak to is high pressure and heat. Even with a 20 point increase in stability cost it wouldn't fix the problem of it being given to every last crew member incredibly early before a threat has even emerged.

I want genetics to work with other departments to overcome the flaws, rather than port something even stronger over, that would undoubtedly worsen the issue SA has. This change would encourage more diverse routes for becoming space proof, and would also make EVA storage more important.

**Comment by PowerfulBacon** [comment]
We basically need a rework of genetics to force downsides to good traits. I think putting it on fiery sweat is too much to make it worth taking at all though. Perhaps low pressure resist could give you a gravity movement slowdown so you are slower on the station and low temperature resist could increase your damage taken from burns (lasers, welders etc.)

**Comment by Poivre-Vert** [comment]
More of the point is that you can get other means to travel in space through other departments, rather than being completely self sufficient in this matter, you can make am entire lap around the station as a human with just pressure resist, so the cold damage can be dealt with through other means.

Space adapt is perfectly fine without cold resist, I just didn't want to completely remove that without putting it somewhere, and firey sweat really is a cool mutation that I'd love to see more of.

**Comment by PowerfulBacon** [comment]
nobody will use fiery sweat even with cold resist, its just too bad (Also why would a mutation that sets you on fire make you stronger against the cold)

**Comment by Poivre-Vert** [comment]
If your internal temperature is high enough to light you on fire, even temporarily in space as well, it would be safe to assume you can't be cooled down, space adapt makes no sense to be cold resistant either.

**Comment by itsmeow** [comment]
:key: Unpopular opinion: Space Adapt is good and not powergamey

But actually though, making genetics more useless is not a great solution to genetics having no downsides/being powergamey. Space Adapt as is, is probably better.

Firey Sweat is just really garbage, will make everyone suffer

I just don't like it idk

**Comment by llol111** [comment]
Thinking about it, i actualy like the fundamental idea of neutering space adapt, since its so easy with it to just neuter whats supposed to be one of the biggest challanges in a space game - space.

I agree with the perviously stated takes though on the matter of fire sweat. Fire sweat is awfull. just split it into two mutations like "pressure resitant" and "improved heat conservation" or whatever.

**Comment by Poivre-Vert** [comment]
> Thinking about it, i actualy like the fundamental idea of neutering space adapt, since its so easy with it to just neuter whats supposed to be one of the biggest challanges in a space game - space.
> 
> I agree with the perviously stated takes though on the matter of fire sweat. Fire sweat is awfull. just split it into two mutations like "pressure resitant" and "improved heat conservation" or whatever.

Problem is, you would be still having the strongest mutation in the game avaliable with no downside, I only put cold resistance on firey sweat so that true space proofing wouldn't be avaliable to a lone geneticist, you either work with other departments to make firey sweat not as dangerous to you, or you work with other departments to offset the temperature damage you take in space or negate it.

As it stands it takes around 5-10 minutes to get all the strong mutations in genetics, and then they will hand out space adapt to every single crew member on the station before either cryoing or disappearing for the rest of the round. No other department can trivialize threats as early and quickly as genetics can; not even xenobiology. 

By putting an actual downside to something you can open up more room for experimentation on how to make that thing work with other people in the round. As well as the cold damage in space being negligible, to the point that coffee warms you up enough to go out in space, and even without that it's not too dangerous. I don't want to outright remove low temperature resistance, but I also do not want to turn firey sweat into a all round temperature resistant mutation as that would be a massive buff to it, as the only danger to crew right now is heat, not space.

File: code/game/machinery/computer/dna_console.dm [hunk 1]
```diff
@@ -682,13 +682,13 @@
...
-						CM.forceMove(src)
-						stored_chromosomes += CM
-						to_chat(usr,"<span class='notice'>[capitalize(CM.name)] added to storage.</span>")
+				//if(scanner_occupant?.client)
+					//var/c_typepath = generate_chromosome()
+					//var/obj/item/chromosome/CM = new c_typepath (drop_location())
+					//if(LAZYLEN(stored_chromosomes) < max_chromosomes)
+						//CM.forceMove(src)
+						//stored_chromosomes += CM
+						//to_chat(usr,"<span class='notice'>[capitalize(CM.name)] added to storage.</span>")
```
Comment (PowerfulBacon) [inline code/game/machinery/computer/dna_console.dm:691]: Either fix this (Make it so each gene can only reward a chromosome once), or remove it entirely and add an alternative method of getting chromosomes (unless there already is one, not sure how chromosomes are obtained)

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
I'm still not a huge fan of attaching it to fiery sweat considering how bad it it. Fiery sweat is going to burn off all your clothes which makes it almost entirely impossible to use as a cold reistant trait. Its especially a problem since making the effect stronger makes fire worse rather than better.
Its a bad one to attach it to I think since its more a grief trait than one you would actually consider and will just be really annoying around the board.

Maybe it should get its own unique mutation with some other downside (Make it so the comfortable temperature of the person is lower so they can't stay in the hot station for too long)?

**Comment by PowerfulBacon** [comment]
Closing due to 1 month without changes

**Pull request closed without merging**